### PR TITLE
[FIX][website_event] Make registration form usable in small screens.

### PR DESCRIPTION
**Pull request by yajo**
Description of the issue/feature this PR addresses: Fix the small screen layout for event registrations.

Current behavior before PR: Before this patch, the event registration form was almost impossible to use in a phone.

Desired behavior after PR is merged: Sane form layout both in small and big screens.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you put some screenshots of before/after for clarifying?

**Comment by Yajo** [comment]
This needs a little bit further patching, tomorrow I'll update it. WIP for now.

**Comment by Yajo** [comment]
This is WIP no more.

Let me show the results. This is **before** the patch:

captura de pantalla de 2017-02-14 11-44-12

captura el 2017-02-14 a las 11 44 38

captura el 2017-02-14 a las 11 44 46

As you can see, the mobile one is almost completelly useless.

Now, **after** the patch:

captura de pantalla de 2017-02-14 11-45-28

captura el 2017-02-14 a las 11 45 45

captura el 2017-02-14 a las 11 48 34

Now the buy section has a bigger visual impact, XML templates are structured in a sane and reusable way, and --most important-- you can actually use it with a phone.

**Comment by pedrobaeza** [comment]
This looks better of course. Maybe the blue color is argueable. Why the runbot error?

**Comment by Yajo** [comment]
Seems like I broke a Tour. I fixed it now.

I added the blue colour because this page is full of similar panels, and this way we give some higher visual impact to the registration form, which is the one that converts more directly in money. However it's not the main issue here, so it can be changed if required.

Not sure who to ping here... @Yenthe666?

**Comment by pedrobaeza** [comment]
Uhm, me neither.

**Comment by Yenthe666** [comment]
Quite sure it is @jke-be you need. He is usually here for website and JS stuff.

**Comment by pedrobaeza** [comment]
This is conflicting with current branch. Can you check?

**Comment by Yajo** [comment]
Conflicts were introduced by @mpa-odoo in https://github.com/odoo/odoo/commit/5d4dd98af2672ccdde37d9587653c80a0ec7c661. I solved them now.

His improvements have been kept unless one little detail: [He displayed "Register now" if only free tickets were available and "Order now" if there was any paid ticket in the list](https://github.com/odoo/odoo/commit/5d4dd98af2672ccdde37d9587653c80a0ec7c661#diff-d4523a5be609dc145d4a2533bf9008eaR96). It seems to me that "Register now" is useful for both scenarios, so I'm using only that.

The rest of the logic should stay the same while having an easier to maintain/reuse template and, most important, usable in phones.

**Comment by pedrobaeza** [comment]
@mpa-odoo, maybe you can check also this improvement?

@JKE-be, can you also check this fine one?

File: addons/website_event_sale/views/event_templates.xml [hunk 1]
```diff
@@ -19,90 +19,66 @@
...
-                    </td>
-                    <td>
-                        <t t-if="ticket.price or editable">
-                            <t t-if="(ticket.price-website.get_current_pricelist().currency_id.compute(ticket.price_reduce, event.company_id.sudo().currency_id)) &gt; 1">
-                            <del class="text-danger mr4" style="white-space: nowrap;" t-field="ticket.price" t-options='{
-                              "widget": "monetary",
-                              "from_currency": event.company_id.sudo().currency_id,
-                              "display_currency": website.get_current_pricelist().currency_id
+    <xpath expr="//t[@t-id='tickets']" position="replace">
+        <t t-set="saleable_tickets" t-value="event.event_ticket_ids.filtered(lambda t: not t.is_expired"/>
```
Comment (JKE-be) [inline addons/website_event_sale/views/event_templates.xml]: missing one parenthesis

File: addons/website_event_sale/views/event_templates.xml [hunk 2]
```diff
@@ -19,90 +19,66 @@
...
-            </t>
-            <t t-if="not ticket_buy">
-                Register Now
+                    </select>
+                </t>
             </t>
-        </button>
-    </xpath>
-    <xpath expr="//form[@id='registration_form']" position="attributes">
-        <attribute name="t-if">event.event_ticket_ids</attribute>
```
Comment (JKE-be) [inline addons/website_event_sale/views/event_templates.xml]: why remove it ?

**Comment by JKE-be** [comment]
Landed in master odoo/odoo@e7fdd6d

**Comment by pedrobaeza** [comment]
Thank you very much!! Very appreciated.

I suppose that you have fixed yourself your comments.

**Pull request closed without merging**